tests/sentry/monitors/test_tasks.py

```diff
@@ -1158,3 +1164,149 @@ def test_monitor_task_trigger_partition_tick_skip(dispatch_tasks):
 
     assert dispatch_tasks.call_count == 2
     assert dispatch_tasks.mock_calls[1] == mock.call(now + timedelta(minutes=2))
+
+
+class MonitorDetectBrokenMonitorEnvTaskTest(TestCase):
+    def create_monitor_and_env(self):
+        monitor = Monitor.objects.create(
+            name="test monitor",
+            organization_id=self.organization.id,
+            project_id=self.project.id,
+            type=MonitorType.CRON_JOB,
+            config={
+                "schedule": [1, "day"],
+                "schedule_type": ScheduleType.INTERVAL,
+                "failure_issue_threshold": 1,
+                "max_runtime": None,
+                "checkin_margin": None,
+            },
+        )
+        monitor_environment = MonitorEnvironment.objects.create(
+            monitor=monitor,
+            environment_id=self.environment.id,
+            status=MonitorStatus.OK,
+        )
+        return (monitor, monitor_environment)
+
+    @with_feature("organizations:crons-broken-monitor-detection")
+    def test_creates_broken_detection(self):
+        monitor, monitor_environment = self.create_monitor_and_env()
+
+        first_checkin = MonitorCheckIn.objects.create(
+            monitor=monitor,
+            monitor_environment=monitor_environment,
+            project_id=self.project.id,
+            status=CheckInStatus.ERROR,
+            date_added=django_timezone.now() - timedelta(days=4),
+        )
+        incident = MonitorIncident.objects.create(
+            monitor=monitor,
+            monitor_environment=monitor_environment,
+            starting_checkin=first_checkin,
+            starting_timestamp=first_checkin.date_added,
+            grouphash=hash_from_values([uuid.uuid4()]),
+        )
+
+        for i in range(3, -1, -1):
```

Comment:
Why are they out of order?

Reply:
Part of the test?

Reply:
Yea just part of the test, not technically required just mimicking checkins coming 
3 days ago, 2 days ago, 1 day ago, and today